### PR TITLE
Adding OAUTH support to API client.

### PR DESCRIPTION
**Pull request by kiddinn**
Adding OAUTH support into the API client

### CODE REVIEW

**Comment by kiddinn** [comment]
Thus should be ready for first round of reviews.  We can start doing more testing once this is pushed out, to make sure this works for all use cases.

File: timesketch/views/auth.py [hunk 1]
```diff
@@ -156,6 +167,88 @@ def logout():
...
+        return abort(
+            HTTP_STATUS_CODE_UNAUTHORIZED, 'Request not authenticated.')
+
+    client_id = current_app.config.get('GOOGLE_OIDC_API_CLIENT_ID')
+    if not client_id:
+        return abort(
+            HTTP_STATUS_CODE_BAD_REQUEST,
+            'No OIDC API client ID defined in the configuration file.')
+
+    token_check_url = '{0:s}?access_token={1:s}'.format(TOKEN_URI, token)
```
Comment (berggren) [inline timesketch/views/auth.py]: Please add a comment with the url to the docs you are following here
https://www.oauth.com/oauth2-servers/signing-in-with-google/verifying-the-user-info/

Comment (berggren) [reply timesketch/views/auth.py]: also, add a comment explining what is happening here. Just easier when we need to pick this up a year from now :)

File: api_client/python/timesketch_api_client/client.py [hunk 2]
```diff
@@ -83,33 +128,108 @@ def _set_csrf_token(self, session):
...
-    def _create_session(self, username, password, verify, auth_mode):
+    def _create_oauth_session(self, client_id, client_secret):
+        """Return an OAuth session.
+
+        Args:
+            client_id: The client ID if OAUTH auth is used.
+            client_secret: The OAUTH client secret if OAUTH is used.
+
+        Return:
+            session: Instance of requests.Session.
```
[the remark this answers is not in the document]
Comment (kiddinn) [reply api_client/python/timesketch_api_client/client.py]: yes it is

[the remark this answers is not in the document]
Comment (kiddinn) [reply api_client/python/timesketch_api_client/client.py]: just like a normal one, except that it adds authentication information to it, to be authenticated

File: data/timesketch.conf [hunk 3]
```diff
@@ -98,6 +98,11 @@ GOOGLE_OIDC_ENABLED = False
 GOOGLE_OIDC_CLIENT_ID = None
 GOOGLE_OIDC_CLIENT_SECRET = None
 
+# If you need to authenticate an API client using OIDC you need to create
+# an OAUTH client for "other", or for native applications.
```
[the remark this answers is not in the document]
Comment (kiddinn) [reply data/timesketch.conf]: done

File: timesketch/views/auth.py (hunk 1 above)
Comment (kiddinn) [reply timesketch/views/auth.py]: done

**Comment by kiddinn** [comment]
@berggren PTAL

**Pull request merged**